core/schema/host.go

```diff
@@ -5,60 +5,102 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/containerd/containerd/labels"
 	"github.com/dagger/dagger/core"
-	"github.com/dagger/dagger/core/socket"
+	"github.com/dagger/dagger/dagql"
+	"github.com/dagger/dagger/engine/sources/blob"
+	"github.com/opencontainers/go-digest"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 type hostSchema struct {
-	*APIServer
-
-	host *core.Host
-	svcs *core.Services
+	srv *dagql.Server
 }
 
 var _ SchemaResolvers = &hostSchema{}
 
-func (s *hostSchema) Name() string {
-	return "host"
-}
-
-func (s *hostSchema) Schema() string {
-	return Host
-}
-
-func (s *hostSchema) Resolvers() Resolvers {
-	return Resolvers{
-		"Query": ObjectResolver{
-			"host": PassthroughResolver,
-		},
-		"Host": ObjectResolver{
-			"directory":     ToResolver(s.directory),
-			"file":          ToResolver(s.file),
-			"unixSocket":    ToResolver(s.socket),
-			"setSecretFile": ToResolver(s.setSecretFile),
-			"tunnel":        ToResolver(s.tunnel),
-			"service":       ToResolver(s.service),
-		},
-	}
+func (s *hostSchema) Install() {
+	dagql.Fields[*core.Query]{
+		dagql.Func("host", func(ctx context.Context, parent *core.Query, args struct{}) (*core.Host, error) {
+			return parent.NewHost(), nil
+		}).Doc(`Queries the host environment.`),
+		dagql.Func("blob", func(ctx context.Context, parent *core.Query, args struct {
+			Digest       string `doc:"Digest of the blob"`
+			Size         int64  `doc:"Size of the blob"`
+			MediaType    string `doc:"Media type of the blob"`
+			Uncompressed string `doc:"Digest of the uncompressed blob"`
+		}) (*core.Directory, error) {
+			blobDef, err := blob.LLB(specs.Descriptor{
+				MediaType: args.MediaType,
+				Digest:    digest.Digest(args.Digest),
+				Size:      args.Size,
+				Annotations: map[string]string{
+					labels.LabelUncompressed: args.Uncompressed, // TODO ???
```

Comment:
```suggestion
					labels.LabelUncompressed: args.Uncompressed, // uncompressed label is required to be set by buildkit's GetByBlob implementation
```

Reply:
For extra context: https://github.com/sipsma/buildkit/blob/e08224fa5567ad24f66b5ed6ead04c5c318ec86e/cache/manager.go#L1664-L1664

This is because buildkit's worker cache will deduplicate cache refs for layers that have different compression (and thus different digests) but the same unpacked contents (and thus the same uncompressed digest).